<ui version="4.0" >
 <class>GenerateDialog</class>
 <widget class="QDialog" name="GenerateDialog" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>772</width>
    <height>460</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>Generate Message Files</string>
  </property>
  <widget class="QLabel" name="label_8" >
   <property name="geometry" >
    <rect>
     <x>10</x>
     <y>10</y>
     <width>751</width>
     <height>81</height>
    </rect>
   </property>
   <property name="text" >
    <string>When you click on "Generate" this will create or re-create message classes, message ID declarations, as well as message handler declarations and implementations.
Inside the message registry file the mapping for message IDs to message handlers is done.
A server module class represents each meassage group. The module class contains a handler implementation for each message of the message group it represents (unless the generate server handler flag is set to no for a given message). The module files will use a prefix in their file name which you can set below. Missing module files will be created.</string>
   </property>
   <property name="alignment" >
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
   </property>
   <property name="wordWrap" >
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget" >
   <property name="geometry" >
    <rect>
     <x>10</x>
     <y>100</y>
     <width>754</width>
     <height>212</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_2" >
    <item>
     <layout class="QGridLayout" name="gridLayout" >
      <item row="0" column="0" >
       <widget class="QLabel" name="label" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Preferred" hsizetype="Preferred" >
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text" >
         <string>Message IDs: Server</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2" >
       <widget class="QLineEdit" name="leMessageID" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
          <horstretch>4</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item row="0" column="3" >
       <widget class="QPushButton" name="btnBrowseID" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text" >
         <string>Choose ...</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" >
       <widget class="QLabel" name="label_2" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Preferred" hsizetype="Preferred" >
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text" >
         <string>Message Registry</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2" >
       <widget class="QLineEdit" name="leMessageReg" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
          <horstretch>4</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item row="2" column="3" >
       <widget class="QPushButton" name="btnBrowseReg" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text" >
         <string>Choose ...</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0" >
       <widget class="QLabel" name="label_4" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Preferred" hsizetype="Preferred" >
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text" >
         <string>Module Classes (Path)</string>
        </property>
       </widget>
      </item>
      <item row="5" column="2" >
       <widget class="QLineEdit" name="leModulePath" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
          <horstretch>4</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item row="5" column="3" >
       <widget class="QPushButton" name="btnBrowseModulePath" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text" >
         <string>Choose ...</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" >
       <widget class="QLabel" name="label_6" >
        <property name="text" >
         <string>Message IDs: Client/Java</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2" >
       <widget class="QLineEdit" name="leMessageID_Client" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
          <horstretch>4</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item row="1" column="3" >
       <widget class="QPushButton" name="btnBrowseID_Client" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text" >
         <string>Choose ...</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0" >
       <widget class="QLabel" name="label_5" >
        <property name="text" >
         <string>Message Classes (Path)</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2" >
       <widget class="QLineEdit" name="leMessageClassesPath" />
      </item>
      <item row="3" column="3" >
       <widget class="QPushButton" name="btnBrowseClassesPath" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text" >
         <string>Choose ...</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="layoutWidget1" >
   <property name="geometry" >
    <rect>
     <x>10</x>
     <y>320</y>
     <width>753</width>
     <height>131</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout" >
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout" >
      <item>
       <widget class="QLabel" name="label_7" >
        <property name="text" >
         <string>Prefix for Modules</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="leModulePrefix" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text" >
         <string>Module</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer" >
        <property name="orientation" >
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType" >
         <enum>QSizePolicy::Expanding</enum>
        </property>
        <property name="sizeHint" stdset="0" >
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <spacer name="verticalSpacer" >
      <property name="orientation" >
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0" >
       <size>
        <width>751</width>
        <height>18</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3" >
      <item>
       <spacer name="horizontalSpacer_2" >
        <property name="orientation" >
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0" >
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="btnCancel" >
        <property name="text" >
         <string>Cancel</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnGenerate" >
        <property name="text" >
         <string>Generate</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <tabstops>
  <tabstop>leMessageID</tabstop>
  <tabstop>btnBrowseID</tabstop>
  <tabstop>leMessageID_Client</tabstop>
  <tabstop>btnBrowseID_Client</tabstop>
  <tabstop>leMessageReg</tabstop>
  <tabstop>btnBrowseReg</tabstop>
  <tabstop>leMessageClassesPath</tabstop>
  <tabstop>btnBrowseClassesPath</tabstop>
  <tabstop>leModulePath</tabstop>
  <tabstop>btnBrowseModulePath</tabstop>
  <tabstop>leModulePrefix</tabstop>
  <tabstop>btnCancel</tabstop>
  <tabstop>btnGenerate</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
